folio-spring-system-user/src/main/java/org/folio/spring/service/SystemUserService.java

```diff
@@ -73,14 +76,33 @@ public SystemUser getAuthedSystemUser(String tenantId) {
    * @return token value
    */
   public UserToken authSystemUser(SystemUser user) {
-    var token = getTokenWithExpiry(user);
+    UserCredentials userCredentials = new UserCredentials(user.username(), systemUserProperties.password());
+    var token = getTokenWithExpiry(userCredentials);
     if (!isValidUserToken(token)) {
       log.info("Login with expiry end-point returned null");
-      return getTokenLegacy(user);
+      return getTokenLegacy(userCredentials);
     }
     return token;
   }
 
+  public UserToken authSystemUser(String tenantId, String username, String password) {
+    log.info("Attempting to issue token for system user [tenantId={}]", tenantId);
+    try (var fex =
+             new FolioExecutionContextSetter(contextBuilder.forSystemUser(prepareSystemUser(tenantId, username)))) {
+      UserCredentials userCredentials = new UserCredentials(username, password);
+      var token = getTokenWithExpiry(userCredentials);
+      if (!isValidUserToken(token)) {
+        log.info("Login with expiry end-point returned null");
+        return getTokenLegacy(userCredentials);
+      }
```

Comment:
This block since it is repeated in the other method should be placed in its own method and re-used. Something like this:

```
  private UserToken getToken(UserCredentials userCredentials) {
    var token = getTokenWithExpiry(userCredentials);
    if (!isValidUserToken(token)) {
      log.info("Login with expiry end-point returned null");
      return getTokenLegacy(userCredentials);
    }
    return token;
  }
```

Reply:
Changes done. please have a look.